pinot-common/src/main/java/org/apache/pinot/common/utils/TlsUtils.java

```diff
@@ -232,8 +235,7 @@ public static void installDefaultSSLSocketFactory(String keyStoreType, String ke
       Protocol.registerProtocol("https",
           new Protocol(CommonConstants.HTTPS_PROTOCOL, new PinotProtocolSocketFactory(sc.getSocketFactory()), 443));
 
-      // FileUploadDownloadClient
-      FileUploadDownloadClient.installDefaultSSLContext(sc);
+      _sslContext = sc;
```

Comment:
I think this should call `setSslContext()`, and that method should be synchronized, or use an `AtomicReference< SSLContext>`, to avoid potential concurrency issues.